### PR TITLE
Update Exec.php

### PR DESCRIPTION
**Pull request by corrilan**
Update to allow multiple paths to be specified during import from server, separated by comma.

### CODE REVIEW

**Comment by ildyria** [comment]
Seems like you managed to get a Pull Request going. If you have any feedback with regard to https://lycheeorg.github.io/docs/contributions.html#illustrated-example please do tell so we can improve.

File: app/Actions/Import/Exec.php
Comment (nagmat84) [inline app/Actions/Import/Exec.php]:
```suggestion
		$paths = explode(';', $path_list);
```

As @ildyria said, `;` would be a much more common choice.

File: app/Actions/Import/Exec.php
Comment (nagmat84) [inline app/Actions/Import/Exec.php]:
```suggestion
		array $paths,
```

or

```suggestion
		string $path,
```

I would like to see an proper array of paths here. "Exploding" a list of paths into an array (as below) should - if at all - happen outside of the method. Alternatively, the method might stay as is and be called iteratively for each path by the caller (this is by `FromServer`).

On top, I would actually go one step further and say it is the responsibility of the front-end to already pass a proper JSON array of paths to be imported and not a string separated by semicolons. I.e. the `ImportServerRequest` should not expect a string for `path` but an array for `paths`. This will help to implement other clients in the long run. Please stop the "stringification" of arguments.

**Review by nagmat84 — CHANGES_REQUESTED** [review]
Please ensure that the client (CLI or web front-end) already passes a correct array of paths to be imported and not a single string separated by semicolons. There might be cases where the semicolon is part of a path and thus must not be used as separator.

File: app/Actions/Import/Exec.php
Comment (corrilan) [reply app/Actions/Import/Exec.php]: Hi @nagmat84, @ildyria originally suggested a comma: "Comma separated is usually a more natural option."

Comment (nagmat84) [reply app/Actions/Import/Exec.php]: OK, maybe, then I got that wrong. Anyway, I would expect either an array as input parameter or even call the method iteratively with a single path. See my  comment below wrt. to line 168.

Comment (ildyria) [reply app/Actions/Import/Exec.php]: As per [this](https://superuser.com/a/748264), I would say screw you if you use comma or semi-colon in your path. :smile:

Comment (corrilan) [reply app/Actions/Import/Exec.php]: Thanks for your input folks, I will get onto this but may be a while, will chip away at weekends, got hands full with a newborn at moment - Lychee is proving great for sharing the photos with the family.

Comment (d7415) [reply app/Actions/Import/Exec.php]: @ildyria Awful source aside, well designed tools allow escaping.

**Comment by ildyria** [comment]
I did a path on that with associated PR: https://github.com/LycheeOrg/Lychee-front/pull/308

File: app/Actions/Import/Exec.php
Comment (ildyria) [reply app/Actions/Import/Exec.php]: Fixed.

File: app/Locale/German.php
Comment (qwerty287) [inline app/Locale/German.php]:
```suggestion
			'UPLOAD_IMPORT_SERVER_INSTR' => 'Importiert alle Fotos, Ordner und Unterordernern in den folgenden absoluten Pfaden (auf dem Server). Mehrere Pfade können mit Leerzeichen getrennt werden, mit \\ können Sie ein Leerzeichen im Pfad verwenden.',
			'UPLOAD_ABSOLUTE_PATH' => 'Absolute Pfade zu Verzeichnissen, mit Leerzeichen getrennt',
```

File: app/Actions/Import/FromServer.php
Comment (qwerty287) [inline app/Actions/Import/FromServer.php:34]: Why do we echo brackets here to get a valid JSON response? Just asking, because it's a bit weird to print something in the action

Comment (nagmat84) [reply app/Actions/Import/FromServer.php:34]: Because it is a streamed response. For a streamed response STDOUT is redirected to the body of the HTTP response. The `echo` and the `$exec->do` are part of an anonymous function which generates the stream. The brackets `[` and `]` are the beginning and end of a JSON array. `$exec->do` prints output as well and creates the entries within the JSON array.

Somewhere else (I believe it is in the action handler for uploads) it is a bit more explicit. There we have a static function instead of an anonymous function which is called `streamGenerator`. Maybe we should do that here as well.

**Review by nagmat84 — APPROVED** [review]
Untested, but looks sane.

**Comment by ildyria** [comment]
I have not tested it either. :innocent:

File: app/Locale/German.php
Comment (nagmat84) [reply app/Locale/German.php]:
```suggestion
			'UPLOAD_IMPORT_SERVER_INSTR' => 'Importiert alle Fotos, Ordner und Unterordner in den folgenden absoluten Pfaden (auf dem Server). Mehrere Pfade können mit Leerzeichen getrennt werden; mit \\ können Sie ein Leerzeichen im Pfad verwenden.',
			'UPLOAD_ABSOLUTE_PATH' => 'Absolute Pfade zu Verzeichnissen, mit Leerzeichen getrennt',
```

I corrected the spelling mistake "Unterordernern" --> "Unterordner".

Moreover, I recognized that we use the formal form to address a user "Sie" (for the Frenchies: vous) instead of the informal form "Du" ("tu"). I know we are very inconsistent about that. This inconsistency has been bothering me for quite a while now, but I haven't felt like fixing that on a global scale.

But we should establish a rule what we want to use and then stick to it:

 - Option 1: Don't address the user directly at all and resort to different alternatives like passive voice, etc. This avoids the problem for languages which have more than a single "you" for everything. Moreover we avoid the problem of deciding whether the user is male or female. (Again for languages which differentiates this aspect.) Something like
   
   > Do you want to delete the photo? This cannot be undone.
   
   would become
   
   > Shall the photo be deleted? This cannot be undone.
   
   Pro: No problem with "Du"/"tu" vs. "Sie"/"vous"; no problem with gender. Con: Passive voice sometimes leads the less legible phrases.

 - Option 2a: Directly address the user in a formal way. Use "Sie"/"vous". 
 
 - Option 2b: Directly address the user in an informal way. Use "Du"/"tu".

**Comment by nagmat84** [comment]
@ildyria Before hitting the "apply" button the next time, read the comment below the suggestion ;-) https://github.com/LycheeOrg/Lychee/pull/1153#discussion_r931281253 The first suggestion contained a spelling error.

File: app/Locale/German.php
Comment (ildyria) [reply app/Locale/German.php]: > Pro: No problem with "Du"/"tu" vs. "Sie"/"vous"; no problem with gender. Con: Passive voice sometimes leads the less legible phrases.

In the same style, my PhD Supervisor wrote his full PhD in passive voice (:scream:). Personally, I went for the "we" as _the authors of the publications_ rather than "we" as _the reader and the author going through the manuscript_.

That being said, I would be going for option 2. Most confirmation windows are directed at the user. e.g. in Debian when doing `apt dist-upgrade`: `Do you want to continue? [Y/n]`

Comment (nagmat84) [reply app/Locale/German.php]: That is fine for English. But there was an option 2a vs. 2b. Formal or informal?

Comment (ildyria) [reply app/Locale/German.php]: FYI, the French translations are making use of the Formal form.

Comment (qwerty287) [reply app/Locale/German.php]: German uses formal too (at least the manpages of GNU coreutils).

**Comment by ildyria** [comment]
Need to merge Lychee-front (waiting for other PR before) and then we can merge this.

**Pull request merged**
